Docs: E2E tests instructions (#92)
os-scar/overlay: e2e/e2e.md

# End-to-End Testing with Playwright

## Creating a test

1. Go to the right e2e file and add a test with specific name.

> 2. create the test with minimal overhead(less locators, less API calls etc.).

Reviewer: ```suggestion
2. create the test with minimal overhead (fewer locators, fewer API calls, etc).
```

## Run and Debug a test

1. download playwright test for vscode extension

> 2. You will see a small green play button near the test you created or want to run.

Reviewer: I don't see a small green play button, maybe it is because we are using custom `test` in `fixture.js`?

Author: ![image](https://github.com/os-scar/overlay/assets/109076747/64307f1f-cb45-40ea-96ba-8beaa16037a0)

Reviewer: Not in my environment 🤷‍♂️

> 1. download playwright test for vscode extension
> 2. You will see a small green play button near the test you created or want to run.
> 3. If the test failed you need to debug it.
> 4. Go to the fixtures file and remove the --headless=new from args(this will run the test on a browser in your computer), you can also uncomment the slowMo for seeing each step.
> 5. run in your terminal yarn e2e:debug it will open a playwright modal with a browser so you can see and debug each step.
> 6. fix the problems and run also in headless mode(undo step 4)

Reviewer: ```suggestion
1. download the [Playwright Test for VSCode](https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright) extension.
2. You will see a small green play button near the test you created or want to run.
3. If the test failed you need to debug it.
4. Go to the fixtures file and remove the `--headless=new` from args (this will run the test on a browser on your computer), you can also uncomment the `slowMo` for seeing each step.
5. Run in your terminal `yarn e2e:debug`. It will open a Playwright modal with a browser so you can see and debug each step.
6. Fix the problems and run also in headless mode (undo step 4)
```

## Problems?

> Ask around, check the Playwright docs.
> https://playwright.dev/docs/intro
> 
> Found an issue? Did we miss a case?
> 
> talk to us https://github.com/os-scar/overlay/

Reviewer: ```suggestion
Ask around, and check the [Playwright docs](https://playwright.dev/docs/intro).

Found an issue? Did we miss a case?

You're welcome to open an issue or talk with us on [Discord](https://discord.gg/H2SMbzh5).
```
